app/src/main/java/org/stepik/android/remote/auth/interceptor/AuthInterceptor.kt

```diff
@@ -0,0 +1,185 @@
+package org.stepik.android.remote.auth.interceptor
+
+import android.content.Context
+import android.webkit.CookieManager
+import com.facebook.login.LoginManager
+import com.vk.sdk.VKSdk
+import okhttp3.Interceptor
+import okhttp3.Response
+import org.stepic.droid.analytic.Analytic
+import org.stepic.droid.configuration.Config
+import org.stepic.droid.core.ScreenManager
+import org.stepic.droid.core.StepikLogoutManager
+import org.stepic.droid.preferences.SharedPreferenceHelper
+import org.stepic.droid.util.AppConstants
+import org.stepic.droid.util.DateTimeHelper
+import org.stepic.droid.util.RWLocks
+import org.stepic.droid.util.addUserAgent
+import org.stepic.droid.web.FailRefreshException
+import org.stepic.droid.web.UserAgentProvider
+import org.stepik.android.remote.auth.model.OAuthResponse
+import org.stepik.android.remote.auth.service.EmptyAuthService
+import org.stepik.android.remote.auth.service.OAuthService
+import org.stepik.android.remote.base.CookieHelper
+import org.stepik.android.view.injection.qualifiers.AuthService
+import org.stepik.android.view.injection.qualifiers.SocialAuthService
+import retrofit2.Call
+import timber.log.Timber
+import java.io.IOException
+import javax.inject.Inject
+
+class AuthInterceptor
+@Inject
+constructor(
+    private val sharedPreference: SharedPreferenceHelper,
+    private val stepikLogoutManager: StepikLogoutManager,
+    private val screenManager: ScreenManager,
+    private val config: Config,
+    private val context: Context,
+    private val userAgentProvider: UserAgentProvider,
+    private val cookieHelper: CookieHelper,
+    private val emptyAuthService: EmptyAuthService,
+    private val analytic: Analytic,
+
+    @SocialAuthService
+    private val socialAuthService: OAuthService,
+
+    @AuthService
+    private val authService: OAuthService
+) : Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        var request = chain.addUserAgent(userAgentProvider.provideUserAgent())
+        try {
+            RWLocks.AuthLock.readLock().lock()
```

Comment:
Why here is used lock from `RWLocks` but in `AuthRemoteDataSourceImpl` is used `AuthLock`?